cql3/statements/select_statement.cc

```diff
@@ -262,6 +264,9 @@ uint64_t select_statement::do_get_limit(const query_options& options, ::shared_p
         if (l <= 0) {
             throw exceptions::invalid_request_exception("LIMIT must be strictly positive");
         }
+        if (!supports_large_paging_state && l > std::numeric_limits<int32_t>::max()) {
+            throw exceptions::invalid_request_exception("LIMIT value out of range");
```

Comment:
A more informative error message would be better. Maybe sth like "Cluster does not support 64-bit LIMIT value: <actual value>"

Reply:
If we're abandoning manually setting the 64-bit limit, there is nothing to throw here anymore

Reply:
Correct